### PR TITLE
Convert parts of example app to Kotlin

### PR DESCRIPTION
**Pull request by aforcier**
As a follow-up to #566, converts more pieces of the example app to Kotlin.

This PR is a bit more of a showcase piece, there are probably some style/idiom conventions we might want to debate here without blocking Kotlin support for the rest of FluxC.

~#566 should be merged first, but reviewing that PR and this one in parallel is probably a good idea.~

### CODE REVIEW

File: example/src/main/java/org/wordpress/android/fluxc/example/AccountFragment.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        (activity.application as ExampleApp).component.inject(this)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View?
+            = inflater.inflate(R.layout.fragment_account, container, false)
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        account_settings.setOnClickListener { changeAccountSettings() }
```
Comment (maxme) [inline example/src/main/java/org/wordpress/android/fluxc/example/AccountFragment.kt]: 🎉

Comment (aforcier) [reply example/src/main/java/org/wordpress/android/fluxc/example/AccountFragment.kt]: Right? 😍

File: example/src/main/java/org/wordpress/android/fluxc/example/MainFragment.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        }
+    }
+
+    @Suppress("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onAuthenticationChanged(event: OnAuthenticationChanged) {
+        if (event.isError) {
+            prependToLog("Authentication error: " + event.error.type)
+
+            when (event.error.type) {
```
Comment (maxme) [inline example/src/main/java/org/wordpress/android/fluxc/example/MainFragment.kt]: This is not easy to read, I wonder if we could better style it.

File: example/src/main/java/org/wordpress/android/fluxc/example/ThreeEditTextDialog.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.app.DialogFragment
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.widget.EditText
+
+class ThreeEditTextDialog : DialogFragment() {
```
Comment (maxme) [inline example/src/main/java/org/wordpress/android/fluxc/example/ThreeEditTextDialog.kt]: Would be great to try [Anko commons](https://github.com/Kotlin/anko/wiki/Anko-Commons-%E2%80%93-Dialogs#alerts) and try to replace this. Next PR ;)?

**Comment by maxme** [comment]
Looks good :shipit:

File: example/src/main/java/org/wordpress/android/fluxc/example/ThreeEditTextDialog.kt (hunk 3 above)
Comment (aforcier) [reply example/src/main/java/org/wordpress/android/fluxc/example/ThreeEditTextDialog.kt]: Agreed - I didn't want to do too much at once. Let's use Anko next.

**Pull request merged**